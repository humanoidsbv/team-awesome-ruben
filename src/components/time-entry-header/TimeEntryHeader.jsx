import React from 'react';
import PropTypes from 'prop-types';

import styles from './timeEntryHeader.module.css';

const TimeEntryHeader = ({ startTimeStamp, stopTimeStamp }) => (
  <div className={styles.container}>
    <p className={styles.date}>{startTimeStamp.slice(0, 10)}</p>
    <p className={styles.date}>08:00:00</p>
  </div>
);

TimeEntryHeader.propTypes = {
  startTimeStamp: PropTypes.string.isRequired,
  stopTimeStamp: PropTypes.string.isRequired
};

export default TimeEntryHeader;
