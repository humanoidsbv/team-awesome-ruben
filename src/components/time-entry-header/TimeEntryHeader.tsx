import React from 'react';

import styles from './TimeEntryHeader.module.css';

interface TimeEntryHeaderProps {
  startTimestamp: string;
  stopTimestamp: string;
}

const TimeEntryHeader = ({
  startTimestamp,
  stopTimestamp
}: TimeEntryHeaderProps): React.ReactElement => {
  const durationHours =
    (Date.parse(stopTimestamp) - Date.parse(startTimestamp)) / 1000 / 60 / 60;

  const formatDuration = (hours: number): string =>
    `${Math.trunc(hours)}:${Math.round((hours % 1) * 60)
      .toString()
      .padStart(2, '0')}`;

  const durationString = formatDuration(durationHours);

  const dateString = new Date(startTimestamp).toLocaleDateString('en-US', {
    day: 'numeric',
    month: 'long',
    weekday: 'long'
  });

  return (
    <div className={styles.container}>
      <p className={styles.date}>{dateString}</p>
      <p className={styles.date}>{durationString}</p>
    </div>
  );
};

export default TimeEntryHeader;
