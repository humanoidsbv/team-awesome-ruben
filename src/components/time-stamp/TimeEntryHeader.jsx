import React from 'react';
import styles from './timeEntryHeader.module.css';

const TimeEntryHeader = () => (
  <div className={styles.container}>
    <p className={styles.date}>Vrijdag 29-07 (Today)</p>
    <p className={styles.date}>08:00:00</p>
  </div>
);

export default TimeEntryHeader;
