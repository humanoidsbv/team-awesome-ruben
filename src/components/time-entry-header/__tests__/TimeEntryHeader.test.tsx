import * as React from 'react';
import { shallow } from 'enzyme';

import TimeEntryHeader from '../TimeEntryHeader';

it('matches component output with expected output', () => {
  const timeEntryHeader = shallow(
    <TimeEntryHeader
      startTimestamp={'2019-09-23T07:00:00.000Z'}
      stopTimestamp={'2019-09-23T15:30:00.000Z'}
    />
  );
  expect(timeEntryHeader).toMatchSnapshot();
});
