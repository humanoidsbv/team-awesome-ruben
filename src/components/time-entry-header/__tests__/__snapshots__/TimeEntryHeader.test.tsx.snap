// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches output with  snapshot 1`] = `
<div
  className="container"
>
  <p
    className="date"
  >
    Monday, September 23
  </p>
  <p
    className="date"
  >
    8:30
  </p>
</div>
`;
